src/xref2/ref_tools.ml

```diff
@@ -817,13 +817,19 @@ let resolve_reference =
   fun env r ->
     match r with
     | `Root (name, `TUnknown) -> (
-        let identifier id = Ok (`Identifier (id :> Identifier.t), None) in
+        let identifier ?text id = Ok (`Identifier (id :> Identifier.t), text) in
         env_lookup_by_name Env.s_any name env >>= function
         | `Module (_, _) as e -> resolved (M.of_element env e)
         | `ModuleType (_, _) as e -> resolved (MT.of_element env e)
         | `Value (id, _) -> identifier id
         | `Type (id, _) -> identifier id
-        | `Label (id, _) -> identifier id
+        | `Label (id, _) ->
+            let text =
+              match Env.lookup_by_id Env.s_label id env with
+              | Some (`Label (_, lbl)) -> Some lbl.Component.Label.text
+              | None -> None
+            in
+            identifier ?text id
```

Comment:
Did this last commit fix a bug?

Reply:
The interface for `Ref_tools.resolve_reference` was changed to also return a `paragraph option`: the "paragraph"[^1] is a possible replacement text computed when resolving the reference. In the case of a title, this "paragraph" is the title.

[^1]:  `paragraph` here is just a type alias for `inline_element with_location list`. Not an actual paragraph.

However, before the last commit, this paragraph was not always output as `Some` by `resolve_reference`, so there was some [code](https://github.com/ocaml/odoc/pull/1116/files#diff-be0d75e85b8a996a6f315280c9f08df1ff70f8af28810af26cf282b28bc7f9aeL232-L238) in `Link` which looked for the title content in the missing case.

In short, the last commit does not fix a bug, it just moves this lookup of title content from `Link.comment_inline_element` to `Ref_tools.resolve_reference` so that the lookup of the title content is centralized in one place.